dashboard/app/models/user.rb

```diff
@@ -731,7 +731,15 @@ def advertised_scripts
   def in_progress_and_completed_scripts
     backfill_user_scripts if needs_to_backfill_user_scripts?
 
-    user_scripts.compact
+    user_scripts.compact.reject do |user_script|
+      begin
+        user_script.script.nil?
```

Comment:
Can this DB query be avoided by doing a lookup in the script cache for `user_script.script_id`?